### PR TITLE
List: add "seek to next" function

### PR DESCRIPTION
**Pull request by frasertweedale**
A common use case is to seek to the next occurrence in the list of
an element matching some predicate (e.g. next unread email).  Add
the 'listSeekToElement' function to satisfy this use case.

'listSeekToElement' is lazy.  Also redefine 'listMoveToElement' in
terms of 'listSeekToElement', making it lazy.  This changes the
constraints on 'listMoveToElement' from (Traversable t) to
(Foldable t, Splittable t), which is reasonable.

### CODE REVIEW

**Comment by frasertweedale** [comment]
@jtdaugherty as usual, please state your preferred function name and I will update the PR :)

File: src/Brick/Widgets/List.hs [hunk 1]
```diff
@@ -545,18 +545,34 @@ splitClamp l i =
...
+-- the beginning.  If no matching element is found, leave the list
+-- unmodified.
+--
+-- /O(n)/.  Only evaluates as much of the container as needed.
+listSeekToElement :: (Foldable t, Splittable t)
+                  => (e -> Bool)
+                  -> GenericList n t e
+                  -> GenericList n t e
+listSeekToElement test l =
+  let
```
Comment (jtdaugherty) [inline src/Brick/Widgets/List.hs]: Please use four-space indents and write `let` bindings as written elsewhere in this module.

**Comment by jtdaugherty** [comment]
Thanks for the patch! I left one comment, but other than that, I think I'd like the new function to be called `listFindBy`.

**Comment by frasertweedale** [comment]
@jtdaugherty thank you; patch updated.

**Comment by jtdaugherty** [comment]
Thank you!

**Comment by jtdaugherty** [comment]
This is released in `0.45`.

**Pull request merged**